netebpfext/net_ebpf_ext_hook_provider.c

```diff
@@ -484,6 +513,10 @@ net_ebpf_extension_hook_provider_register(
 
     status = NmrRegisterProvider(characteristics, local_provider_context, &local_provider_context->nmr_provider_handle);
     if (!NT_SUCCESS(status)) {
+
+        // The docs don't mention the (out) handle status on failure, so explicitly mark it as invalid.
```

Comment:
Sure they do.
`typedef _Return_type_success_(return >= 0) LONG NTSTATUS;`
means that NTSTATUS is documented via the `_Return_type_success_` annotation as saying that on a failure, any Out fields cannot be assumed to be initialized.  It's ok to initialize them anyway but I don't think it's correct to say the docs don't mention it.

Reply:
Ah. I was not aware of that.  I was referring to the more 'traditional' doc [source](https://learn.microsoft.com/en-us/windows-hardware/drivers/ddi/netioddk/nf-netioddk-nmrregisterprovider).  It does not mention the SAL enhancements.

Reply:
Fixed.